sql/driver/link_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein (tamird@gmail.com)
+
+package driver
+
+import (
+	"go/build"
+	"regexp"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/util/leaktest"
+)
+
+var remotePkgRegxp = regexp.MustCompile("[^/]+\\.[^/]+/")
+
+func TestNoLinkForbidden(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	if build.Default.GOPATH == "" {
+		t.Skip("GOPATH isn't set")
+	}
+
+	imports := make(map[string]struct{})
+
+	var addImports func(string)
+	addImports = func(root string) {
+		pkg, err := build.Import(root, build.Default.GOPATH, 0)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, imp := range pkg.Imports {
+			// https: //github.com/golang/tools/blob/master/refactor/importgraph/graph.go#L115
+			if imp == "C" {
+				if remotePkgRegxp.MatchString(pkg.ImportPath) {
+					t.Errorf("sql/driver depends on cgo via %s", pkg.ImportPath)
+				}
+				continue // "C" is fake
+			}
+			if _, ok := imports[imp]; !ok {
+				imports[imp] = struct{}{}
```

Comment:
We could use `pkg.ImportPath` as the value in this map so that it could tell you one (arbitrary) edge in the graph that led to the forbidden package.